### DISCUSSION TITLE
Prevent adding working hours

### DISCUSSION THREAD

**Discussion by sevyb** [craigk5n/webcalendar#517] [Q&A]
Hello. I have defined working hours from 8:00-18:00. When I create an entry at 8:00 and add a duration, the calendar automatically adds 2 hours before 8:00 starting at 6:00. How can I prevent this behaviour? I want to keep 8:00-18:00.

**Comment by catelluxembourg** [comment]
Hello,
Take look in your user timezone settings

**Comment by sevyb** [comment]
Thanks for the hint, this was not the problem. Actually, I found out now, that on individual calenders this does not happen but only on views I create when I enter an appointment directly on the view. Anyone who could help?

**Reply by catelluxembourg** [reply]
hello,
I insist: all users must have the same timezone if you are in Belgium for example. I have already encountered this.

**Comment by mjanser** [comment]
The issue might be in the following code: https://github.com/craigk5n/webcalendar/blob/402a92a533d450fd18876ddb845c7e5f2c20f728/view_r.php#L353-L358

The method `getTime()` seems to return the time in UTC. This resets the first slot, but the event is displayed in a later slot because of a different timezone.

**Reply by mjanser** [reply]
I changed the above-mentioned code to the following, which seems to help.
```php
      if( $ev[$j]->getDuration() > 0 && $ev[$j]->getDuration() != 1440 ) {
        $localTime = str_replace ( ':', '', display_time ( $ev[$j]->getDatetime(), 0, '', '24' ) );
        $slot = calc_time_slot( $localTime . '00', false );
        if ( $slot < $first_slot ) {
          $first_slot = $slot;
        }
      }
``` 
Not sure though if this is the correct and best solution.
